lualib/resty/events/broker.lua

```diff
@@ -37,36 +37,35 @@ end
 local function broadcast_events(self, unique, data)
     local n = 0
 
-    -- if unique, schedule to a random worker
-    local idx = unique and random(1, nkeys(self._clients))
+    local queues = self._queues
 
-    for _, q in pairs(self._clients) do
-
-        -- skip some and broadcast to one workers
-        if unique then
-            idx = idx - 1
-
-            if idx > 0 then
-                goto continue
-            end
-        end
-
-        local ok, err = q:push(data)
+    local first_worker_id = self._first_worker_id
+    local last_worker_id = self._last_worker_id
 
+    if unique then
+        -- if unique, schedule to a random worker
+        local worker_id = random(first_worker_id, last_worker_id)
+        local worker_queue = queues[worker_id]
+        local ok, err = worker_queue:push(data)
         if not ok then
             log(ERR, "failed to publish event: ", err, ". ",
                      "data is :", cjson_encode(decode(data)))
+        end
 
-        else
-            n = n + 1
+        n = n + 1
 
-            if unique then
-                break
+    else
+        for worker_id = first_worker_id, last_worker_id do
+            local worker_queue = queues[worker_id]
+            local ok, err = worker_queue:push(data)
+            if not ok then
+                log(ERR, "failed to publish event: ", err, ". ",
```

Comment:
same comment about logging the worker id

Reply:
added.